### DISCUSSION TITLE
How Does Rain (archive db) Work?

### DISCUSSION THREAD

**Discussion by tomdotorg** [bellrichm/WeeWX-MQTTSubscribe#150] [Q&A] [answered]
Hi,

New to this stuff, but working on getting a functional replacement for a WeatherLink-like setup using WifiLogger2. Here are the rain fields sent on every loop:

<img width="156" alt="image" src="https://user-images.githubusercontent.com/52498/162410015-582db1a8-bbcf-421a-ae25-c3abc2caba0a.png">

What would be the proper mapping in weewx.conf to capture (eventually) the actual correct rain for generated archive records? Does this question even make sense?

This is what I am doing (I copied the config with the intention of tweaking it) and this is clearly not right - it's raining today and my weewx shows like 20" because of my wrong mappings..

```        
        [[[[rain15]]]]
            ignore = false # opt back in for this field
            name = rain
        [[[[rainr]]]]
            ignore = false # opt back in for this field
            name = rainRate
```
edit: correct data (WeatherLink) : http://vtwx.tom.org/
and incorrect MQTT as (incorrectly) configured above : https://vtwx-dev.tom.org/

**Comment by bellrichm** [comment, marked as answer]
We may want to bounce this by the experts in the [WeeWX Google Group](https://groups.google.com/g/weewx-user). Here is some background information. 

The field `rain` in WeeWX is a delta field. Prior to a 4.x release, WeeWX did not have the concept of a cumulative rain observation and it was the responsibility of the driver or service to convert from cumulative to delta. MQTTSubscribe provides this via a [conntains_total](https://github.com/bellrichm/WeeWX-MQTTSubscribe/wiki/Configuring#contains_total) option. Also see, 
[issue 149](https://github.com/bellrichm/WeeWX-MQTTSubscribe/issues/149)

I couldn't find much documentation on it, but it appears that WeeWX now has a `rainTotal` field. Here is some additional info.
https://github.com/weewx/weewx/issues/491#issuecomment-716040604
https://github.com/weewx/weewx/blob/a55892e1d5cd41167d7eff9635f5a1750ad2a850/bin/weewx/wxxtypes.py#L557

Net, I think this is what you want
```
        [[[[rain15]]]] # assuming this is cumulative for 15 minutes??
            ignore = false # opt back in for this field
            name = rainTotal
```
Not sure which cumulative field makes the most sense...

**Reply by tomdotorg** [reply]
Thanks, Rich. I will construct a question for the weewx group.

**Reply by tomdotorg** [reply]
Quick non-update. I am still reading up on this, but it is slow going as I am not a Python expert. Stay tuned..

**Reply by tomdotorg** [reply]
@bellrichm quick question: is it possibly as easy as choosing one of the cumulative rain fields (rain15, 1h, etc.) and mark it as contains_total? So something like this:

```
        [[[[rain15]]]]
            ignore = false # opt back in for this field
            contains_total = true
            name = rain
```

**Reply by bellrichm** [reply]
Yes, that should work. I was only pointing out that WeeWX now natively supports it. When, I get some time, I'll see if I can figure out how to use the native WeeWX support

**Reply by bellrichm** [reply]
> Quick non-update. I am still reading up on this, but it is slow going as I am not a Python expert. Stay tuned..

If you get stuck, don't hesitate to ask. Questions also help to point out where the documentation needs more work.
